CHANGELOG.md

```diff
@@ -1,5 +1,96 @@
+* Remove unused `chain`, `methodcaller`, `mkdtemp`, `StringIO` imports in `conda.common.compat`; apply other fixes from `ruff --fix .` in the test suite. (#12294)
+* Remove unused optimization for searching packages based on `*[track_features=<feature name>]`. (#12314)
+* Remove Notebook spec support from `conda env`; this was deprecated already and scheduled to be remove in version 4.5. (#12307)
+* Mark `conda_exe` in `context.py` and a topic in `print_package_info` `cli/main_info.py` for official deprecation. (#12276)
+* Marking `conda.utils.hashsum_file` as pending deprecation. Use `conda.gateways.disk.read.compute_sum` instead. (#12414)
+* Marking `conda.utils.md5_file` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "md5")` instead. (#12414)
+* Marking `conda.gateways.disk.read.compute_md5sum` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "md5")` instead. (#12414)
+* Marking `conda.gateways.disk.read.compute_sha256sum` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "sha256")` instead. (#12414)
+* Drop Python 3.7 support. (#12436)
+
+### Docs
+
+* Added docs for `conda.deprecations`. (#12452)
+* Updated some instances of "Anaconda Cloud" to be "Anaconda.org". (#12238)
+* Added documentation on the specifications for `conda search` and `conda install`. (#12304)
+* Mark `conda.utils.safe_open` for deprecation. Use builtin `open` instead. (#12415)
+
+### Other
+
+* Update `<cache key>.json.state` `repodata.json` cache format; check `mtime` against cached `repodata.json`. (#12090)
+* Skip redundant `tar --no-same-owner` when running as root on Linux, since newer `conda-package-handling` avoids setting ownership from the archive. (#12231)
+* Add additional extensions to `conda.common.path` for future use. (#12261)
+* Pass `--cov` in test runner scripts but not in `setup.cfg` defaults, for easier debugability. (#12268)
+* Constrain conda-build to at least >=3.18.3, released 2019-06-20. (#12309)
+* Improve `start.bat` windows development script. (#12311)
+* Provide conda-forge-based Docker images and fix the bundled `minio` binary. (#12335)
+* Add support for conda-forge-based CI runtimes. On Linux (all architectures), unit & integration tests will use Python 3.10. On Windows, Python 3.8. On macOS, only the unit tests are run with conda-forge (_instead_ of `defaults`!), using Python 3.9. (#12350, #12447 via #12448)
+* Fix testing data issue where the `subdir` entry in some files was mismatched. (#12389)
+* Initialize conda after installing test requirements during CI. (#12446)
+* Speedup pre-commit by a factor of 15 by removing ignored hooks (pylint/bandit). This locally reduces the pre-commit runtime from ~43sec to 2.9sec and thus makes it possible to run pre-commit in a loop during development to constantly provide feedback and style the code. (#12466)
```

Comment:
```suggestion
* Update vendored version of `py-cpuinfo` to 0.9.0. (#12319)
* Improved code coverage. (#12346, #12457, #12469)
* Add a note about `use_only_tar_bz2` being enabled on `PackagesNotFoundError` exceptions. (#12353)
* Added to conda CLI help that `conda remove -n <myenv> --all` can be used to delete environments. (#12378)
* Handle Python import errors gracefully when loading conda plugins from entrypoints. (#12460)

### Bug fixes

* Fixed errors when renaming without an active environment. (#11915)
* Prevent double solve attempt if `PackagesNotFoundError` is raised. (#12201)
* Virtual packages follow `context.subdir` instead of `platform.system()` to enable cross-platform installations. (#12219)
* Don't export `__glibc` virtual package when `CONDA_OVERRIDE_GLIBC=""`. (#12267)
* Fix `arg_parse` pass-through for `--version` and `--help` in `conda.xsh`. (#12344)
* Filter out `None` path values from `pwd.getpwall()` on Unix systems, for users without home directories, when running as root. (#12063)

### Deprecations

* Drop `toolz.itertoolz.unique` in favor of custom `conda.common.iterators.unique` implementation. (#12252)
* Stop using `OrderedDict`/`odict` since `dict` preserves insert order since Python 3.7. (#12254)
* Mark `conda._vendor.boltons` for deprecation in 23.9.0. (#12272, #12482)
* Mark `conda_exe` in `context.py` and a topic in `print_package_info` `cli/main_info.py` for official deprecation. (#12398)
* Remove unused `chain`, `methodcaller`, `mkdtemp`, `StringIO` imports in `conda.common.compat`; apply other fixes from `ruff --fix .` in the test suite. (#12294)
* Remove unused optimization for searching packages based on `*[track_features=<feature name>]`. (#12314)
* Remove Notebook spec support from `conda env`; this was deprecated already and scheduled to be remove in version 4.5. (#12307)
* Mark `conda_exe` in `context.py` and a topic in `print_package_info` `cli/main_info.py` for official deprecation. (#12276)
* Marking `conda.utils.hashsum_file` as pending deprecation. Use `conda.gateways.disk.read.compute_sum` instead. (#12414)
* Marking `conda.utils.md5_file` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "md5")` instead. (#12414)
* Marking `conda.gateways.disk.read.compute_md5sum` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "md5")` instead. (#12414)
* Marking `conda.gateways.disk.read.compute_sha256sum` as pending deprecation. Use `conda.gateways.disk.read.compute_sum(path, "sha256")` instead. (#12414)
* Drop Python 3.7 support. (#12436)

### Docs

* Added docs for `conda.deprecations`. (#12452)
* Updated some instances of "Anaconda Cloud" to be "Anaconda.org". (#12238)
* Added documentation on the specifications for `conda search` and `conda install`. (#12304)
* Mark `conda.utils.safe_open` for deprecation. Use builtin `open` instead. (#12415)

### Other

* Update `<cache key>.json.state` `repodata.json` cache format; check `mtime` against cached `repodata.json`. (#12090)
* Skip redundant `tar --no-same-owner` when running as root on Linux, since newer `conda-package-handling` avoids setting ownership from the archive. (#12231)
* Add additional extensions to `conda.common.path` for future use. (#12261)
* Pass `--cov` in test runner scripts but not in `setup.cfg` defaults, for easier debugging. (#12268)
* Constrain conda-build to at least >=3.18.3, released 2019-06-20. (#12309)
* Improve `start.bat` Windows development script. (#12311)
* Provide conda-forge-based Docker images and fix the bundled `minio` binary. (#12335)
* Add support for conda-forge-based CI runtimes. On Linux (all architectures), unit & integration tests will use Python 3.10. On Windows, Python 3.8. On macOS, only the unit tests are run with conda-forge (_instead_ of `defaults`!), using Python 3.9. (#12350, #12447 via #12448)
* Fix testing data issue where the `subdir` entry in some files was mismatched. (#12389)
* Initialize conda after installing test requirements during CI. (#12446)
* Speedup pre-commit by a factor of 15 by removing ignored hooks (`pylint`/`bandit`). This locally reduces the pre-commit runtime from ~43sec to 2.9sec and thus makes it possible to run pre-commit in a loop during development to constantly provide feedback and style the code. (#12466)
```

Reply:
(kept this all in one comment in order to reduce the need for multiple commits)

Reply:
Resolved in https://github.com/conda/conda/pull/12486/commits/0c576521c66cf3e8b9b5b11a6c311c5743580160